pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -568,34 +568,57 @@ func (r *ReconcileLogStorage) Reconcile(ctx context.Context, request reconcile.R
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileLogStorage) elasticsearchSecrets(ctx context.Context) ([]*corev1.Secret, error) {
+// Checks if public cert secret managed by ECK is valid. If not valid, the cert is deleted. Returns true if the cert was deleted.
+func (r *ReconcileLogStorage) deleteInvalidECKManagedPublicCertSecret(ctx context.Context, secret *corev1.Secret, certKey string, svcDNSNames []string) (bool, error) {
```

Comment:
I suggest this function returns only `error`. Error should mean fail to delete. If the secret is not present, it should return a custom error.

Reply:
Fixed